tests/test_queuing.py

```diff
@@ -406,22 +344,119 @@ def __init__(self, *args):
                 self.queue()
 
             def queue(self):
-                qml.QueuingContext.append(self, owns=tuple(self.obs))
+                QueuingContext.append(self, owns=tuple(self.obs))
 
                 for o in self.obs:
                     try:
-                        qml.QueuingContext.update_info(o, owner=self)
+                        QueuingContext.update_info(o, owner=self)
                     except AttributeError:
                         pass
 
                 return self
 
-        with qml._queuing.AnnotatedQueue() as q:
+        with AnnotatedQueue() as q:
             A = qml.PauliZ(0)
             B = qml.PauliY(1)
             tensor_op = AnnotatingTensor(A, B)
 
         assert q.queue == [A, B, tensor_op]
-        assert q.get_info(A) == {"owner": tensor_op}
-        assert q.get_info(B) == {"owner": tensor_op}
-        assert q.get_info(tensor_op) == {"owns": (A, B)}
+        assert q._get_info(A) == {"owner": tensor_op}
+        assert q._get_info(B) == {"owner": tensor_op}
+        assert q._get_info(tensor_op) == {"owns": (A, B)}
+
+
+class TestOperationRecorder:
+    """Test the OperationRecorder class."""
+
+    def test_circuit_integration(self):
+        """Tests that the OperationRecorder integrates well with the
+        core behaviour of PennyLane."""
+        expected_output = (
+            "Operations\n"
+            + "==========\n"
+            + "PauliY(wires=[0])\n"
+            + "PauliY(wires=[1])\n"
+            + "RZ(tensor(0.4, requires_grad=True), wires=[0])\n"
+            + "RZ(tensor(0.4, requires_grad=True), wires=[1])\n"
+            + "CNOT(wires=[0, 1])\n"
+            + "\n"
+            + "Observables\n"
+            + "===========\n"
+        )
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(a, b, c):
+            qml.RX(a, wires=0)
+            qml.RY(b, wires=1)
+
+            with qml.tape.OperationRecorder() as recorder:
```

Comment:
Why are we nesting into a `QNode` here which is also evaluated at the end of the test? 🤔

Reply:
I do not know, I didn't modify these tests aside from changing the import location of the operation recorder 😆

Reply:
GitHub is treating it as an entire addition for some reason.